interfaces/builtin/unity8_calendar.go

```diff
@@ -0,0 +1,130 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"github.com/snapcore/snapd/interfaces"
+)
+
+const unity8CalendarPermanentSlotAppArmor = `
+# Description: Allow operating as the EDS service. Reserved because this
+#  gives privileged access to the system.
+# Usage: reserved
+
+# DBus accesses
+#include <abstractions/dbus-session-strict>
+dbus (bind)
+	bus=session
+	name="org.gnome.evolution.dataserver.Calendar7",
+dbus (bind)
+	bus=session
+	name=org.gnome.evolution.dataserver.Subprocess.Backend.*,
+
+########################
+# Calendar
+########################
+dbus (receive, send)
+	bus=session
+	path=/org/gnome/evolution/dataserver/CalendarFactory,
+dbus (receive, send)
+	bus=session
+	path=/org/gnome/evolution/dataserver/CalendarFactory
+	interface=org.gnome.evolution.dataserver.CalendarFactory,
+dbus (receive, send)
+	bus=session
+	path=/org/gnome/evolution/dataserver/CalendarFactory
+	interface=org.freedesktop.DBus.*,
+
+dbus (receive, send)
+	bus=session
+	path=/org/gnome/evolution/dataserver/CalendarView/**,
+dbus (receive, send)
+	bus=session
+	path=/org/gnome/evolution/dataserver/CalendarView/**
+	interface=org.gnome.evolution.dataserver.CalendarView,
+dbus (receive, send)
+	bus=session
+	path=/org/gnome/evolution/dataserver/CalendarView/**
+	interface=org.freedesktop.DBus.*,
+
+dbus (receive, send)
+	bus=session
+	path=/org/gnome/evolution/dataserver/Subprocess/Backend/Calendar/**
+	interface=org.gnome.evolution.dataserver.Subprocess.Backend,
+dbus (receive, send)
+	bus=session
+	path=/org/gnome/evolution/dataserver/Subprocess/Backend/Calendar/**
+	interface=org.freedesktop.DBus.*,
+
+dbus (receive, send)
+	bus=session
+	path=/org/gnome/evolution/dataserver/Subprocess/**
+	interface=org.gnome.evolution.dataserver.Calendar,
+dbus (receive, send)
+	bus=session
+	path=/org/gnome/evolution/dataserver/Subprocess/**
+	interface=org.freedesktop.DBus.*,
+`
```

Comment:
I think nearly all of these should be moved to ConnectedSlotAppArmor. In snappy, we mediate *both* what slot implementations the client can connect to (ConnectedPlug) and what clients the slot implementation can connect to (ConnectedSlot).  PermanentSlot policy should only contain enough policy so that the service may start listening for clients. As such, please rework these rules into ConnectedSlotAppArmor, using `###PLUG_SECURITY_TAGS###`. There are plenty of examples in interfaces/builtin/* to get you started.